src/copilot/app/rag/rag_processor.py

```diff
@@ -0,0 +1,155 @@
+from rag.prompts import OPENAI_RAG_SYSTEM_PROMPT_DE
+from rag.models import RAGRequest, EmbeddingRequest
+
+from autocomplete.queries import semantic_similarity_match
+from utils.embedding import get_embedding
+
+
+class RAGProcessor:
+    """
+    Class implementing the RAG process
+
+    Parameters
+    ----------
+    model : str
+        LLM Model used for generating chatbot answers
+    max_token : int
+    stream : bool
+    temperature : float
+    top_p : float
+    top_k : int
+    client
+    """
+    def __init__(self, model: str, max_token: int, stream: bool, temperature: float,
+                 top_p: float, top_k: int, client):
+        self.model = model
+        self.max_tokens = max_token
+        self.stream = stream
+        self.temperature = temperature
+        self.top_p = top_p
+        self.k_retrieve = top_k
+
+        self.client = client
+
+    async def retrieve(self, request: RAGRequest, language: str = None, k: int = 0):
+        """
+        Retrieve context documents related to the user input question.
+
+        Only supports retrieval of 1 document at the moment (set in /config/config.yaml).
+
+        .. todo::
+            multi-doc retrieval later
+
+        Parameters
+        ----------
+        request : RAGRequest
+            User input question
+        language : str
+            Question and context documents language
+        k : int, default 0
+            Number of context documents to return (need to be revised, current logic in the code is confusing)
+        """
+        k = self.k_retrieve if k is None else k
+
+        rows = await semantic_similarity_match(request.query, db_name='embeddings', language=language, k=k)
+        documents = [dict(row) for row in rows][0]
+
+        return {"contextDocs": documents["text"], "sourceUrl": documents["url"], "cosineSimilarity": documents["distance"]}
```

Comment:
change 

- return {"contextDocs": documents["text"], "sourceUrl": documents["url"], "cosineSimilarity": documents["distance"]} 

to

- return {"contextDocs": documents["text"], "sourceUrl": documents["url"], "similarity_metric": documents["similarity_metric"]}